test/common/config/filesystem_subscription_impl_test.cc

```diff
@@ -18,7 +18,10 @@ namespace Config {
 namespace {
 
 class FilesystemSubscriptionImplTest : public testing::Test,
-                                       public FilesystemSubscriptionTestHarness {};
+                                       public FilesystemSubscriptionTestHarness {
+public:
+  static void SetUpTestSuite() { Filesystem::fileSystemForTest().setUseMemfiles(false); }
```

Comment:
I'm wondering how this pattern of updating a static will work for coverage tests where we link everything together.

Another pattern that might work well is to make a test-mixin a la TestUsingSimulatedTime: https://github.com/envoyproxy/envoy/blob/7852c7b28c51e725e6fbe864bcdc3ce19def3936/test/test_common/simulated_time_system.h#L130 

it would touch the same files you did; just via the inheritance chain rather than SetUpTestSuite.

Reply:
coverage build seems happy, so I think we're good here unless you have a straight up prefrence.

Reply:
Sorry for erroneous ping - github pebkac fail.

I don't think we have the problem with coverage any more.  We could instead set if it runs or not via scoped thing, so it unsets at the end of a test run.  WDYT?

Reply:
setting it with a scoped thing works for me!